extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../../../src/services';
+
+/**
+ * @param {!JsonObject} media
+ * @param {AmpElement} apesterElement
+ * @param {!JsonObject} consentObj
+ * @return {Promise}
+ */
+export function handleCompanionVideo(media, apesterElement, consentObj) {
+  const monetizationSettings = media['campaignData'] || {};
+  const companionCampaignOptions =
+    monetizationSettings['companionCampaignOptions'] || {};
+  const companionRawSettings = monetizationSettings['companionOptions'] || {};
+  if (
```

Comment:
The function would be cleaner if it had early bail outs (guard clauses) rather than nested if conditions.

Reply:
Hi @sparhami we followed your changes, and keep talked and handle the other issues with @lannka , can you take a look on this PR again?